internal/transport/grpchttp2/http2bridge_test.go

```diff
@@ -0,0 +1,468 @@
+		t.Errorf("WriteRSTStream(): SettingID: got %d, want %d", errCode, ErrCodeProtocol)
+	}
+}
+
+func (s) TestBridge_WriteSettings(t *testing.T) {
+	c := &testConn{}
+	f := NewFramerBridge(c, c, 0)
+	f.WriteSettings(Setting{ID: SettingsHeaderTableSize, Value: 200})
+
+	if errs := checkWrittenHeader(c.wbuf[0:9], 6, FrameTypeSettings, 0, 0); len(errs) > 0 {
+		t.Errorf("WriteSettings():\n%s", errors.Join(errs...))
+	}
+
+	if settingID := readUint16(c.wbuf[9:11]); settingID != uint16(SettingsHeaderTableSize) {
+		t.Errorf("WriteSettings(): SettingID: got %d, want %d", settingID, SettingsHeaderTableSize)
+	}
+	if settingVal := readUint32(c.wbuf[11:15]); settingVal != 200 {
+		t.Errorf("WriteSettings(): SettingVal: got %d, want %d", settingVal, 200)
+	}
+}
+
+func (s) TestBridge_WriteSettingsAck(t *testing.T) {
+	c := &testConn{}
+	f := NewFramerBridge(c, c, 0)
+	f.WriteSettingsAck()
+
+	if errs := checkWrittenHeader(c.wbuf[0:9], 0, FrameTypeSettings, FlagSettingsAck, 0); len(errs) > 0 {
+		t.Errorf("WriteSettingsAck():\n%s", errors.Join(errs...))
+	}
+}
+
+func (s) TestBridge_WritePing(t *testing.T) {
+	c := &testConn{}
+	f := NewFramerBridge(c, c, 0)
+	acks := []bool{true, false}
+	wantData := [8]byte{1, 2, 3, 4, 5, 6, 7, 8}
+	for _, ack := range acks {
+		f.WritePing(ack, wantData)
+		var flag Flag
+		if ack {
+			flag |= FlagPingAck
+		}
+		if errs := checkWrittenHeader(c.wbuf[0:9], 8, FrameTypePing, flag, 0); len(errs) > 0 {
+			t.Errorf("WritePing():\n%s", errors.Join(errs...))
+		}
+		data := c.wbuf[9:]
+		for i := range data {
+			if data[i] != wantData[i] {
+				t.Errorf("WritePing(): Data[%d]: got %d, want %d", i, data[i], wantData[i])
+			}
+		}
+		c.wbuf = c.wbuf[:0]
+	}
+}
+
+func (s) TestBridge_WriteGoAway(t *testing.T) {
+	c := &testConn{}
+	f := NewFramerBridge(c, c, 0)
+	f.WriteGoAway(2, ErrCodeFlowControl, []byte("debug_data"))
+
+	if errs := checkWrittenHeader(c.wbuf[0:9], 18, FrameTypeGoAway, 0, 0); len(errs) > 0 {
+		t.Errorf("WriteGoAway():\n%s", errors.Join(errs...))
+	}
+
+	if lastStream := readUint32(c.wbuf[9:13]); lastStream != 2 {
+		t.Errorf("WriteGoAway(): LastStreamID: got %d, want %d", lastStream, 2)
+	}
+	if code := ErrCode(readUint32(c.wbuf[13:17])); code != ErrCodeFlowControl {
+		t.Errorf("WriteGoAway(): Code: got %d, want %d", code, ErrCodeFlowControl)
+	}
+	if data := string(c.wbuf[17:]); data != "debug_data" {
+		t.Errorf("WriteGoAway(): Data: got %q, want %q", data, "debug_data")
+	}
+}
+
+func (s) TestBridge_WriteWindowUpdate(t *testing.T) {
+	c := &testConn{}
+	f := NewFramerBridge(c, c, 0)
+	f.WriteWindowUpdate(1, 2)
+
+	if errs := checkWrittenHeader(c.wbuf[0:9], 4, FrameTypeWindowUpdate, 0, 1); len(errs) > 0 {
+		t.Errorf("WriteWindowUpdate():\n%s", errors.Join(errs...))
+
+	}
+	if inc := readUint32(c.wbuf[9:13]); inc != 2 {
+		t.Errorf("WriteWindowUpdate(): Inc: got %d, want %d", inc, 2)
+	}
+}
+
+func (s) TestBridge_WriteContinuation(t *testing.T) {
```

Comment:
And for all of your tests, please add top-level comment of the form:

// Tests <describe the scenario in short> and verifies <describe the validation in short>

This would make it easier for someone reading the test to know at a high level what is happening in the test before jumping in to read the actual test. Thanks.

Reply:
Done.